paseos/visualization/space_animation.py

```diff
@@ -0,0 +1,280 @@
+import pykep as pk
+import numpy as np
+from dotmap import DotMap
+from typing import List
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+from matplotlib.artist import Artist
+from loguru import logger
+
+from paseos.actors.base_actor import BaseActor
+from paseos.actors.spacecraft_actor import SpacecraftActor
+from paseos.actors.ground_station_actor import GroundstationActor
+from paseos.paseos import PASEOS
+from paseos.visualization.animation import Animation
+
+
+class SpaceAnimation(Animation):
+    """This class visualizes the central body, local actor and known actors over time."""
+
+    def __init__(self, sim: PASEOS) -> None:
+
+        super().__init__(sim)
+        logger.trace("Initializing animation")
+        # Create list of objects to be plotted
+        current_actors = self._make_actor_list(sim)
+        self._norm_coeff = self._local_actor._central_body.radius
+
+        local_time_d = self._local_actor.local_time.mjd2000
+        for known_actor in current_actors:
+            pos, _ = known_actor.get_position_velocity(pk.epoch(local_time_d))
+            pos_norm = [x / self._norm_coeff for x in pos]
+            self.objects.append(DotMap(actor=known_actor, positions=np.array(pos_norm)))
+
+        # create figure
+        self.fig = plt.figure(figsize=plt.figaspect(0.5) * 1.5)
+        self.ax = plt.subplot(projection="3d")
+        self.ax.set_title(self._sec_to_ddhhmmss(local_time_d / pk.SEC2DAY))
+        self.ax.get_xaxis().set_ticks([])
+        self.ax.get_yaxis().set_ticks([])
+        self.ax.get_zaxis().set_ticks([])
+        self.n_trajectory = 50  # how many samples from histories to visualize
+        self._textbox_offset = 0.25  # how many samples from histories to visualize
+
+        # plot the objects
+        self._plot_central_body()
+        self._plot_actors()
+
+        plt.show(block=False)
```

Comment:
I am not sure if this specific call is the problem, but atm for the me it doesn't work neither in jupyter nor as a script.

In Jupyter running the example, nothing happens at all
In the script a window opens the flickers, doesn't show anything and then closes again

Reply:
Jupyter interactive plotting is a bit more involved. I made an example for Jupyter now.